src/js/plugins/menu_toggle/plugin.menu_toggle.js

```diff
@@ -1,113 +1,194 @@
-      $menuTab.css('display', 'block');
       removeEventHandlers(br);
+      br.showNavigation();
     }
   
+    /**
+     * Removes click handlers on elements that house the book pages
+     *
+     * @param { object } br - BookReader instance
+     */
     var removeEventHandlers = function removeEventHandlers(br) {
       if (br.refs.$brPageViewEl) {
-        br.refs.$brPageViewEl[0].removeEventListener('click', toggleMenuIfCenterClick, true);
+        br.refs.$brPageViewEl[0].removeEventListener('click', onBookClick, true);
       }
       if (br.refs.$brTwoPageView) {
-        br.refs.$brTwoPageView[0].removeEventListener('click', toggleMenuIfCenterClick, true);
+        br.refs.$brTwoPageView[0].removeEventListener('click', onBookClick, true);
       }
     }
 
+    /**
+     * Toggle functionality
+     * Responsible for calling native functions `hideNavigation` & `showNavigation`
+     * Makes sure only 1 toggle action is taken at a time using `togglingNav` switch.
+     *
+     * @params { object } br - bookreader instance
+     */
+    var togglingNav = false; /* flag to make sure animations only fire once */
     var toggleNav = function toggleNav(br) {
+      if (togglingNav) {
+        return;
+      }
+
+      togglingNav = true;
+      const navToggled = function navToggled() {
+        togglingNav = false;
+        window.removeEventListener('BookReader:navToggled', navToggled);
+      };
+      $(document).on('BookReader:navToggled', navToggled);
+
       var menuIsShowing = br.navigationIsVisible();
       if (menuIsShowing) {
         br.hideNavigation();
       } else {
         br.showNavigation();
       }
     }
-  
-    var toggleMenuIfCenterClick = function toggleMenuIfCenterClick(br, e) {
-      var book = e.currentTarget;
-      if (!book) {
-        return;
-      }
+
+    /**
+     * Confirms whether or not the click happened in the nav toggle zone
+     *
+     * @param { object } event - JS event object
+     */
+    var isCenterClick = function isCenterClick(event) {
+      var book = event.currentTarget;
       var bookWidth = book.offsetWidth;
       var leftOffset = book.offsetLeft
       var bookEndPageFlipArea = Math.round(bookWidth / 3);
       var leftThreshold = Math.round(bookEndPageFlipArea + leftOffset); // without it, the click area is small
       var rightThreshold = Math.round(bookWidth - bookEndPageFlipArea + leftOffset);
-      var isCenterClick = (e.clientX > leftThreshold) && (e.clientX < rightThreshold);
-      
-      if (isCenterClick) {
-        toggleNav(br);
-        event.stopPropagation();
-      }
+      var isCenterClick = (event.clientX > leftThreshold) && (event.clientX < rightThreshold);
+
+      return isCenterClick;
     }
 
-    function registerEventHandlers(br) {
+    /**
+     * Confirms whether or not the click happened in the background
+     *
+     * @param { object } event - JS event object
+     */
+    var isBackground = function isBackground(event) {
```

Comment:
I might have this method take the `target` directly instead of the event since it's only checking for `event.target` inside of it.

Reply:
good call 👍